### ISSUE TITLE
Add city name to stop

### ISSUE THREAD

**Issue by AltNico** [grote/osm2gtfs#108]
When [generating national GTFS feeds](https://github.com/grote/osm2gtfs/milestone/1) it is important to have the possibility to add the city name to a stop because it is very likely that one stop name exists in various cities.

Example: *Managua, Mercado Mayoreo* (Bus stop *Mercado Mayoreo* in Managua, Nicaragua)

**Comment by grote** [comment]
At least Navitia should add the city automatically to stops based on information from OSM. If you add this to stop names, you would see the city twice in Transportr.

**Comment by AltNico** [comment]
In Transportr Nicaragua city names are missing from the stop names, so for example I only see 'Hospital'. Do I need to configure anything anywhere in order to automatically add that name?

**Comment by prhod** [comment]
All the GTFS i encountered doesn't specify city names in stop names. I don't think adding this information is a god idea if it's a workaround specific to navitia. 
Just in case, there is also the difficulty of specifying the city level to take into account, that can be different from a country from an other, and with exceptions for some cities (example of London in UK).

**Comment by nlehuby** [comment]
I agree : I don't think adding city to stop name is a good GTFS practice.
In my opinion, this could be handled in a custom creator if really needed

**Comment by ialokim** [comment]
> AFAIK it needs administrative boundaries with admin_level = 8 (cities) to build streets and addresses.

@prhod @nlehuby Is is possible to automatically add the city name to stops?

**Comment by prhod** [comment]
@ialokim I'm not sure to understand your question. Are you asking this in navitia or in osm2gtfs ? 
In Navitia, the city construction is automatic but has to be enhanced. In osm2gtfs, I don't think it's a good idea to add the city name to stops.

**Comment by ialokim** [comment]
I was asking about navitia, sorry, my fault.

> In Navitia, the city construction is automatic

So can this city name automatically be appended to stop names? Or can it be accessed within the API when requesting stop details?

**Comment by prhod** [comment]
To keep it simple, Navitia loads cities from OSM with the boundaries. Each point (stop, Point of Interest or address) is associated with `administrative_regions`. The `label`of the object gives a preformated text with the stop name and the city name. The raw stop name and the properties of the `adminstrative_regions` can also be accessed.
There could be several in case of stop areas grouping stop_points of different cities, for example : https://canaltp.github.io/navitia-playground/play.html?request=https%3A%2F%2Fapi.navitia.io%2Fv1%2Fcoverage%2Ffr-idf%2Fstop_areas%2Fstop_area%3AOIF%3ASA%3A59%3A3681487
Tell me if you want any more details ;-)

**Comment by ialokim** [comment]
Okay perfect, thanks a lot for this explanation! So definitly there's no need to add city names to stops inside _osm2gtfs_.

Testing it [on our own navitia instance](https://navitia.mapanica.net/v1/coverage/ni-esteli/stop_points), it seems the label is always equal to the name. We've included OSM data for Nicaragua. Do we need some extra settings and would it work on navitia.io anyway or is it a problem with the OSM data?

Looking at[ _pte_'s source code](https://github.com/schildbach/public-transport-enabler/blob/master/enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java#L224), I'm fearing this `label` field isn't supported there, so probably we should open an issue or PR over there.

**Comment by grote** [comment]
PTE should already take the city of the first `administrative_region` and add it to the `place` variable. The label of the object with the preformated text is not what you want in PTE.

**Comment by ialokim** [comment]
> PTE should already take the city of the first administrative_region and add it to the place variable. The label of the object with the preformated text is not what you want in PTE.

Looking at the code, it seems it only does in some cases (hardcoded, in German language), see [here](https://github.com/schildbach/public-transport-enabler/blob/master/enabler/src/de/schildbach/pte/dto/Location.java#L139). That would definitely not work for Nicaragua.

**Comment by grote** [comment]
I am afraid, you are looking at the wrong code. Check this: https://github.com/schildbach/public-transport-enabler/blob/master/enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java#L244

**Comment by ialokim** [comment]
Okay, my fault, the `place` variable is _public_, but it isn't used by the [`uniqueShortName()`](https://github.com/schildbach/public-transport-enabler/blob/master/enabler/src/de/schildbach/pte/dto/Location.java#L146) function which is used inside Transportr. It seems that it could be fixed inside Transportr then by not using the `uniqueShortName` but combining stop and city name [here](https://github.com/grote/Transportr/blob/master/app/src/main/java/de/grobox/transportr/utils/TransportrUtils.kt#L64).

**Comment by grote** [comment]
It is used by Transportr for location auto-completion.

I suggest you inspect the API results of your instance directly to see if administrative regions exist there.

**Comment by ialokim** [comment]
> It is used by Transportr for location auto-completion.

Okay, perfect.

> I suggest you inspect the API results of your instance directly to see if administrative regions exist there.

[We already know that they are not](#issuecomment-355780231), but I wanted to be sure they could be shown by Transportr. Thanks a lot for your explanations!

**Comment by ialokim** [comment]
I think this could be closed.

**Comment by AltNico** [comment]
@grote Can you give Mikolai the permission to close issues? I can do it but he can't.
